packages/theme/src/cli/utilities/theme-environment/hot-reload/server.ts

```diff
@@ -110,15 +116,18 @@ export function setupInMemoryTemplateWatcher(ctx: DevServerContext) {
       })
     } else {
       // Unknown files outside of assets. Wait for sync and reload:
-      onSync(() => triggerHotReload(fileKey, ctx))
+      onSync(() => {
+        triggerHotReload(fileKey, ctx)
+      })
     }
   }
 
-  const handleFileUnlink = ({fileKey, onSync}: ThemeFSEventPayload<'unlink'>) => {
-    const extension = extname(fileKey)
-    if (isAsset(fileKey) && extension === '.liquid') {
+  const handleFileDelete = ({fileKey, onSync}: ThemeFSEventPayload<'unlink'>) => {
+    // Liquid assets are proxied, so we need to wait until the file has been deleted on the server before reloading
+    const isLiquidAsset = isAsset(fileKey) && extname(fileKey) === '.liquid'
+    if (isLiquidAsset) {
       onSync?.(() => {
-        triggerHotReload(fileKey, ctx)
+        triggerHotReload(fileKey.replace('.liquid', ''), ctx)
```

Comment:
Do we handle deletes in the `client.ts` file? Or are we just making a new `v=<timestamp>` and getting a 404?

Reply:
> are we just making a new v=<timestamp> and getting a 404?

If I'm not mistaken, this is what's happening behind the scenes. 

Also, I'm removing the `.liquid` here so that we will trigger a `css` reload for `css.liquid` files rather than a `full` reload

Alternatively, I can move the above logic into`triggerHotReload` instead of replacing the `extname`. wdyt?